### PR TITLE
BOM-2071 : Drop Python 2.7 Support

### PR DESCRIPTION
**Pull request by mzulqarnain1**
Relevant JIRA : https://openedx.atlassian.net/browse/BOM-2071

### CODE REVIEW

**Review by UsamaSadiq — APPROVED** [review]

File: jenkins/pull_request_creator.py [hunk 1]
```diff
@@ -76,12 +73,12 @@ def _create_new_branch(self):
 
     def _create_new_pull_request(self):
         # If there are reviewers to be added, split them into python lists
-        if isinstance(self.user_reviewers, (str, six.text_type)) and self.user_reviewers:
+        if isinstance(self.user_reviewers, (str, str)) and self.user_reviewers:
```
Comment (jmbowman) [inline jenkins/pull_request_creator.py:76]: This now checks to see if it's an instance of `str` or `str`; should switch it from a tuple to just `str`.  Same below.

File: jenkins/repo_health_reports.py [hunk 2]
```diff
@@ -93,12 +90,12 @@ def main(sha, repo_root, repo_name, org, user_reviewers, team_reviewers):
             LOGGER.info("Creating a new pull request")
 
             # If there are reviewers to be added, split them into python lists
-            if isinstance(user_reviewers, (str, six.text_type)) and user_reviewers:
+            if isinstance(user_reviewers, (str, str)) and user_reviewers:
```
Comment (jmbowman) [inline jenkins/repo_health_reports.py:93]: Same in this file.

File: jenkins/tests/test_codecov_analysis.py [hunk 3]
```diff
@@ -22,9 +19,9 @@ def get_pulls(self, state, sort, direction):
         return self.prs
 
 
-class MockPR(object):
+class MockPR:
 
-    def __init__(self, title, commits=[], age=10000):
+    def __init__(self, title, commits=None, age=10000):
```
Comment (jmbowman) [inline jenkins/tests/test_codecov_analysis.py:24]: Maybe just remove the default argument instead?  I think the new default would blow up in `reversed()` below, and we specify it in all calls anyway.

File: jenkins/tests/test_job.py [hunk 4]
```diff
@@ -1,8 +1,8 @@
-from __future__ import absolute_import
+# pylint: disable=missing-module-docstring
+from unittest import TestCase
 
 from mock import patch
```
Comment (jmbowman) [inline jenkins/tests/test_job.py:4]: Do you want to handle the switch to `unittest.mock` in this PR or do that separately?

File: tox.ini [hunk 5]
```diff
@@ -45,7 +45,7 @@ commands =
     make clean
     pycodestyle .
     pylint travis
```
Comment (jmbowman) [inline tox.ini:47]: This is now redundant with the line below.  I think the reason you hit so many linting problems was that the line below had only been running the Python 3-related subset of pylint checks; all the other checks were only being run on this directory.

File: jenkins/tests/test_job.py (hunk 4 above)
Comment (mzulqarnain1) [reply jenkins/tests/test_job.py:4]: That will be done with Python 3.8 compatibility.

File: tox.ini (hunk 5 above)
Comment (mzulqarnain1) [reply tox.ini]: Removed.

**Review by jmbowman — APPROVED** [review]

**Comment by mzulqarnain1** [comment]
@jmbowman Please merge this PR.

**Pull request merged**